pkg/generators/rules/names_match.go

```diff
@@ -104,15 +104,15 @@ func (n *NamesMatch) Validate(t *types.Type) ([]string, error) {
 				continue
 			}
 			jsonName := strings.Split(jsonTag, ",")[0]
-			if !namesMatch(goName, jsonName) {
+			if !GoAndJSONNameMatch(goName, jsonName) {
 				fields = append(fields, goName)
 			}
 		}
 	}
 	return fields, nil
 }
 
-// namesMatch evaluates if goName and jsonName match the API rule
+// GoAndJSONNameMatch evaluates if goName and jsonName match the API rule
```

Comment:
I don't think this should be exported. But if it must be exported, make a new package for it to live in; no one should be importing this rules package IMO, in fact we should move it under an /internal directory.